miner/worker.go

```diff
@@ -1201,60 +1211,131 @@ func (w *worker) commit(uncles []*types.Header, interval func(), update bool, st
 	return nil
 }
 
-func (w *worker) findMostProfitableBundle(bundles []types.Transactions, coinbase common.Address, parent *types.Block, header *types.Header) (types.Transactions, *big.Int, *big.Int, uint64) {
-	maxBundlePrice := new(big.Int)
-	maxTotalEth := new(big.Int)
-	var maxTotalGasUsed uint64
-	maxBundle := types.Transactions{}
+type simulatedBundle struct {
+	txs           types.Transactions
+	mevGasPrice   *big.Int
+	ethToCoinbase *big.Int
+	totalGasUsed  uint64
+}
+
+func (w *worker) generateFlashbotsBundle(bundles []types.Transactions, coinbase common.Address, parent *types.Block, header *types.Header) (simulatedBundle, int, error) {
+	simulatedBundles, err := w.simulateBundles(bundles, coinbase, parent, header)
+	if err != nil {
+		return simulatedBundle{}, 0, err
+	}
+
+	sort.SliceStable(simulatedBundles, func(i, j int) bool {
+		return simulatedBundles[j].mevGasPrice.Cmp(simulatedBundles[i].mevGasPrice) < 0
+	})
+
+	return w.mergeBundles(simulatedBundles, parent, header)
+}
+
+func (w *worker) mergeBundles(bundles []simulatedBundle, parent *types.Block, header *types.Header) (simulatedBundle, int, error) {
+	tailGasPrice := big100Gwei // default to 100 Gwei if can't find good price
+	parentTxs := parent.Transactions()
+	if len(parentTxs) > 0 {
```

Comment:
very unlikely but still:
* an adversary unsigned miner could construct a block with the last transaction with a huge gas price which would prevent all the other miners from accepting MEV bundles in the next round